docs/milestone2_walkthrough.md

```diff
@@ -0,0 +1,167 @@
+# Milestone 2 Walkthrough
+
+## Walkthrough Goals
+
+* Get a local environment running with 1 control plane cluster and 2 workload clusters
+* Define a Gateway in the control plane with a HTTPS listener, and have it instantiated in 2 workload clusters
+* Deploy an Application with a HTTPRoute to the 2 workload clusters, and have the route attached to the Gateway listener
+* Curl the Application host with DNS resolving from Route 53 and a TLS Certificate
+* Apply a Kuadrant RateLimitPolicy to the Application
+* Curl the Application host and see the rate limit being enforced
+
+## Prerequisites
+
+* [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+* [AWS Account](https://aws.amazon.com/) with [Route53](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/Welcome.html) Zone permissions
+    * Choose an existing Hosted zone or create a new one for the walkthrough.
+* openssl>=3
+    * On macos a later version is available with `brew install openssl`. You'll need to update your PATH as macos provides an older version via libressl as well
+    * On fedora use `dnf install openssl`
+
+## Environment Setup
+
+Create a set of environment files with your AWS configuration.
+First, create a file called `aws-credentials.env` in the root of the project and add the following config, replacing values as necessary
+
+```bash
+AWS_ACCESS_KEY_ID=<AWS_ACCESS_KEY_ID>
+AWS_SECRET_ACCESS_KEY=<AWS_SECRET_ACCESS_KEY>
+AWS_REGION=<AWS_REGION>
+```
+
+Next, create a file called `controller-config.env` in the root of the project and add the following config, replacing values as necessary. For example, if your Hosted zone is for the domain `apps.example.com`, use that as the `ZONE_ROOT_DOMAIN`. This will mean your Application can use a hostname of `myapp.apps.example.com`
+
+```bash
+ZONE_ROOT_DOMAIN=<my.hosted.zone.domain.name>
+AWS_DNS_PUBLIC_ZONE_ID=<my.hosted.zone.id>
+```
+
+Start the local clusters and run the controller in the control plane
+
+```bash
+MCTC_WORKLOAD_CLUSTERS_COUNT=2 make local-setup
```

Comment:
im getting this error when running these commands
`ERROR   controller-runtime.metrics      metrics server failed to listen. You may want to disable the metrics server or use another port if it is due to conflicts       {"error": "error listening on :8080: listen tcp :8080: bind: address already in use"}
sigs.k8s.io/controller-runtime/pkg/metrics.NewListener
        /Users/rachellawton/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/metrics/listener.go:48
sigs.k8s.io/controller-runtime/pkg/manager.New
        /Users/rachellawton/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/manager/manager.go:407
main.main
        /Users/rachellawton/Projects/multi-cluster-traffic-controller/cmd/controller/main.go:90
runtime.main
        /opt/homebrew/Cellar/go/1.20.2/libexec/src/runtime/proc.go:250
2023-03-10T14:40:28Z    ERROR   setup   unable to start manager {"error": "error listening on :8080: listen tcp :8080: bind: address already in use"}
main.main
        /Users/rachellawton/Projects/multi-cluster-traffic-controller/cmd/controller/main.go:99
runtime.main
        /opt/homebrew/Cellar/go/1.20.2/libexec/src/runtime/proc.go:250
exit status 1
make: *** [run-controller] Error 1`